### PR TITLE
[Refactor] Fixing mypy issues of `service/dataset`

### PR DESCRIPTION
**Pull request by khoaguin**

### CODE REVIEW

File: packages/syft/src/syft/service/dataset/dataset.py [hunk 1]
```diff
@@ -138,7 +138,9 @@ class Asset(SyftObject):
     __repr_attrs__ = ["name", "shape"]
 
     def __init__(
-        self, description: Optional[Union[MarkdownDescription, str]] = "", **data
+        self,
+        description: Optional[Union[MarkdownDescription, str]] = "",
+        **data: Dict[str, Any],
```
Comment (kiendang) [inline packages/syft/src/syft/service/dataset/dataset.py]:
```suggestion
        **data: Any,
```

`*args` and `**kwargs` should be typed as `*args: Any` and `**kwargs: Any`, not `*args: Tuple[Any]` and `**kwargs: Dict[str, Any]`. see #6791

File: packages/syft/src/syft/service/dataset/dataset.py [hunk 2]
```diff
@@ -329,7 +331,7 @@ class CreateAsset(SyftObject):
     class Config:
         validate_assignment = True
 
-    def __init__(self, description: Optional[str] = "", **data):
+    def __init__(self, description: Optional[str] = "", **data: Dict[str, Any]) -> None:
```
Comment (kiendang) [inline packages/syft/src/syft/service/dataset/dataset.py]:
```suggestion
    def __init__(self, description: Optional[str] = "", **data: Any) -> None:
```

File: packages/syft/src/syft/service/dataset/dataset.py [hunk 3]
```diff
@@ -470,14 +472,16 @@ class Dataset(SyftObject):
     __repr_attrs__ = ["name", "url", "created_at"]
 
     def __init__(
-        self, description: Optional[Union[str, MarkdownDescription]] = "", **data
-    ):
+        self,
+        description: Optional[Union[str, MarkdownDescription]] = "",
+        **data: Dict[str, Any],
```
Comment (kiendang) [inline packages/syft/src/syft/service/dataset/dataset.py]:
```suggestion
        **data: Any,
```

**Review by kiendang — CHANGES_REQUESTED** [review]

File: packages/syft/src/syft/service/dataset/dataset.py (hunk 2 above)
Comment (khoaguin) [reply packages/syft/src/syft/service/dataset/dataset.py]: Thanks @kiendang

**Review by kiendang — APPROVED** [review]

**Pull request merged**